### ISSUE TITLE
WebUI frozen when I2P connections is lost

### ISSUE THREAD

**Issue by pkreuzt** [majestrate/XD#64]
### info

git revision / version: 0.2.2
OS: linux
Architecture: armv7
Running through I2PD

### problem

WebUI doesn't advert of I2P connection lose. When I2pd dies or closes for some reason, the webui shows a frozen state which still reads there are peers:

X peers connected on Y torrents (ratio) , etc

This gives the impression it's waiting to download or upload. There is no indication of the lost connection.

**Comment by majestrate** [comment]
good point, I'll work on adding an rpc method for checking online status this weekend.

**Comment by majestrate** [comment]
fix attempted in https://github.com/majestrate/XD/commit/476938e0d6f39b4174a26b0228749b67e9341add

**Comment by pkreuzt** [comment]
This seems to work, webui no longer freezes and now it shows different behaviour when connection is lost. It also recovers well from reconnection.
Probably this is not the intended solution but it shows a ghost torrent only in "All" tab when connection is lost:

![captura de pantalla_2018-07-02_18-27-16](https://user-images.githubusercontent.com/6936840/42175589-795814d8-7e15-11e8-8262-08750ef36e4e.png)
